src/hooks/useHandleClickOutside.ts

```diff
@@ -0,0 +1,21 @@
+import * as React from "react";
+
+type Props = {
+  handleClickOutside: (e: unknown) => void;
+  clickOutsideToClose: boolean;
+};
+
+export function useHandleClickOutside({
```

Comment:
I don't see the implementation of "clicking outside" here (just a mousedown event),
I think this should be implemented with the usage of ref like below

https://github.com/streamich/react-use/blob/master/docs/useClickAway.md

Reply:
Hmm actually implemented it using `useEvent`, the problem with using `ref` here is that we have two `refs` to handle 😄